### DISCUSSION TITLE
post_233779

### DISCUSSION THREAD

**Discussion by cincuranet** [cincuranet/blog-comments#1319] [Comments]
imported 

https://www.tabsoverspaces.com/id/233779

**Comment by cincuranet** [comment]
Imported comment written by _svick_ on _2019-04-24T12:36:45_:

<p>In the second snippet, I think you forgot to declare the test variable in the using.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2019-04-24T12:47:56_:

<p>Right, thanks. Fixing it right now.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Michael DiSibio_ on _2020-04-22T22:43:08_:

<p>If creating the instance relies on a factory method that might fail, usually in a try/finally block we would add `instance?.Dispose()`, so to avoid having to add a try/finally as well, would there be anything wrong with this syntax...?<br><code><br>var test = null;<br>await using((test = new AwaitUsingConfigureAwaitTest()).ConfigureAwait(false))<br>{<br>  test.Dummy();<br>}<br></code></p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2020-04-23T09:00:41_:

<p>If the factory method failed, there's nothing to dispose.</p>

**Comment by hwfbcisod** [comment]
Thank you! Was just banging my head for 10 minutes trying to figure out how to set ConfigureAwait on that pesky using.
